sstables/mc/parquet_writer.hh

```diff
@@ -0,0 +1,973 @@
+    return parquet4seastar::bytes_view(
+            reinterpret_cast<const parquet4seastar::byte*>(bv.data()),
+            bv.size());
+}
+
+parquet4seastar::bytes_view sstring_to_bytes_view(::sstring s) {
+    return parquet4seastar::bytes_view(
+            reinterpret_cast<const parquet4seastar::byte*>(s.data()),
+            s.size());
+}
+
+constexpr parquet4seastar::logical_type::logical_type
+map_type(abstract_type::kind scylla_type) {
+    using kind = abstract_type::kind;
+    switch (scylla_type) {
+        case kind::ascii:
+            return STRING{};
+        case kind::boolean:
+            return BOOLEAN{};
+        case kind::byte:
+            return UINT8{};
+        case kind::bytes:
+            return BYTE_ARRAY{};
+        case kind::counter:
+            return UNKNOWN{};
+        case kind::date:
+            return BYTE_ARRAY{};
+        case kind::decimal:
+            return BYTE_ARRAY{};
+        case kind::double_kind:
+            return DOUBLE{};
+        case kind::duration:
+            return BYTE_ARRAY{};
+        case kind::empty:
+            return UNKNOWN{};
+        case kind::float_kind:
+            return FLOAT{};
+        case kind::inet:
+            return BYTE_ARRAY{};
+        case kind::int32:
+            return INT32{};
+        case kind::list:
+            return BYTE_ARRAY{};
+        case kind::long_kind:
+            return INT64{};
+        case kind::map:
+            return BYTE_ARRAY{};
+        case kind::reversed:
+            return BYTE_ARRAY{};
+        case kind::set:
+            return BYTE_ARRAY{};
+        case kind::short_kind:
+            return INT16{};
+        case kind::simple_date:
+            return DATE{};
+        case kind::time:
+            return TIME_INT64{false, TIME_INT64::NANOS};
+        case kind::timestamp:
+            return TIMESTAMP{false, TIMESTAMP::MILLIS};
+        case kind::timeuuid:
+            return BYTE_ARRAY{};
+        case kind::tuple:
+            return BYTE_ARRAY{};
+        case kind::user:
+            return BYTE_ARRAY{};
+        case kind::utf8:
+            return STRING{};
+        case kind::uuid:
+            return BYTE_ARRAY{};
+        case kind::varint:
+            return BYTE_ARRAY{};
+        default:
+            return UNKNOWN{};
+    }
+}
+
+constexpr parquet4seastar::format::Type::type
+map_physical_type(abstract_type::kind scylla_type) {
+    parquet4seastar::logical_type::logical_type lt =
+            map_type(scylla_type);
+    parquet4seastar::format::Type::type pt =
+            std::visit([] (const auto& x) { return x.physical_type; }, lt);
+    return pt;
+}
+
+namespace parts {
+enum metadata_parts {
+    HEADER,
+    HEADER_PARTITION_KEY,
+    HEADER_PARTITION_KEY_X,
+    HEADER_DELETION_LDT,
+    HEADER_DELETION_MFDA,
+    SROW,
+    SROW_FLAGS,
+    SROW_EXTENDED_FLAGS,
+    SROW_CELLS,
+    SROW_CELLS_X,
+    SROW_CELLS_X_FLAGS,
+    SROW_CELLS_X_DT,
+    SROW_CELLS_X_DLDT,
```

Comment:
Where can I learn what is LDT, MFDA, DT, DLDT, DTTL and DMFDA? These need at least to be documented (if they are standard Parquet vocabulary, then a link to documentation would suffice), and even better, all TLA's and other types of acronyms should be expanded to its full human-readable form.

Also, what's the difference between HEADER_PARTITION_KEY and HEADER_PARTITION_KEY_X? (same for CELLS, CELLS_X and other _X pairs.

Reply:
Please take a look at 70fe746

> even better, all TLA's and other types of acronyms should be expanded to its full human-readable form.

I'm not sure about that. In fact, that's what I started with, but names like STATIC_ROW_CELLS_X_LOCAL_DELETE_TIME turned out unwieldy.